src/components/TodoList.jsx

```diff
@@ -1,66 +1,239 @@
+                },
+                {
+                    taskName: "last one to do",
+                    date: new Date(),
+                    id: 2,
+                    color: "",
+                    textDecoration: "",
+                    bIsDone: false,
+                    bIsImportant: false
+                }
+            ];
+        for (let task of newTasks) {
+            Object.freeze(task)
+        };
         this.state = {
-            tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
-            ],
-            input: "nichego net"
+            tasks: List(newTasks),
+            // Main input field text value
+            input: "",
+            // Current soring option to sort tasks 
+            sortingOption: "Sort by",
+            // Id of task in edit mode
+            taskInEditId: -1,
+            // Id of last created task
+            lastCreatedId: 2
         };
-
         this.addTask = this.addTask.bind(this);
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this);
+        this.sortTasks = this.sortTasks.bind(this);
+        this.setTaskInEditId = this.setTaskInEditId.bind(this);
+        this.setNewTaskText = this.setNewTaskText.bind(this);
+        this.setTaskColor = this.setTaskColor.bind(this);
+        this.setTaskTextDecoration = this.setTaskTextDecoration.bind(this)
+    }
+    setTaskColor(taskId) {
+        const { tasks } = this.state
+        this.setState({
+            ...this.state,
+            tasks: tasks.map(task => task.id === taskId
+                ? (({
+                    ...task,
+                    bIsImportant: !task.bIsImportant,
+                    color: !task.bIsImportant ? "red" : "black"
+                }))
+                : task
+            )
+        })
+    }
+    // Set text deoration style of task
+    setTaskTextDecoration(taskId) {
+        const { tasks } = this.state
+        this.setState({
+            ...this.state,
+            tasks: tasks.map(task => task.id === taskId
+                ? (({
+                    ...task,
+                    bIsDone: !task.bIsDone,
+                    textDecoration: !task.bIsDone ? "line-through" : ""
+                }))
+                : task
+            )
+        })
     }
 
-    removeTask(text) {
+    // Set text of task in edit mode 
+    setTaskInEditId(todoId) {
+        this.setState({
+            ...this.state,
+            taskInEditId: todoId
+        })
+    }
+    // Change name of task
+    setNewTaskText(newText) {
+        const { tasks, taskInEditId } = this.state
+        this.setState({
+            ...this.state,
+            tasks: tasks.map(task => task.id === taskInEditId
+                ? ({ ...task, taskName: newText })
+                : task),
+            taskInEditId: -1
```

Comment:
better use `null` for when there is no value. Explicitly compare `=== null` to check if there is any edited task. This is cleaner and less error prone, cause **the type of the value here then tells us, whether it's edited or not**. Cause null is its own type.